cassandra-4/src/main/java/io/debezium/connector/cassandra/Cassandra4CommitLogBatchParser.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.cassandra;
+
+import java.util.List;
+
+import org.apache.cassandra.db.commitlog.CommitLogPosition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.connector.base.ChangeEventQueue;
+
+public class Cassandra4CommitLogBatchParser extends Cassandra4CommitLogParserBase {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Cassandra4CommitLogBatchParser.class);
+
+    public Cassandra4CommitLogBatchParser(Cassandra4CommitLogProcessor.LogicalCommitLog commitLog, List<ChangeEventQueue<Event>> queues,
+                                          CommitLogProcessorMetrics metrics, CassandraConnectorContext context) {
+        super(commitLog, queues, metrics, context);
+    }
+
+    @Override
+    public Cassandra4CommitLogProcessor.ProcessingResult parse() {
+        try {
+            parseIndexFile(commitLog);
+
+            while (!commitLog.completed) {
+                if (completePrematurely) {
+                    LOGGER.info("{} completed prematurely", commitLog.toString());
+                    return new Cassandra4CommitLogProcessor.ProcessingResult(commitLog, Cassandra4CommitLogProcessor.ProcessingResult.Result.COMPLETED_PREMATURELY);
```

Comment:
could not you make `COMPLETED_PREMATURELY` as import?

Reply:
done